gc.c

```diff
@@ -2659,6 +2661,23 @@ rb_gc_mark_roots(void *objspace, const char **categoryp)
 
 #define TYPED_DATA_REFS_OFFSET_LIST(d) (size_t *)(uintptr_t)RTYPEDDATA(d)->type->function.dmark
 
+void
+rb_gc_ractor_moved(VALUE dest, VALUE src)
+{
+    void *objspace = rb_gc_get_objspace();
+    st_data_t id = 0;
+    if (UNLIKELY(FL_TEST_RAW(src, FL_SEEN_OBJ_ID))) {
+        id = rb_gc_impl_object_id_move(objspace, dest, src);
```

Comment:
I think `rb_gc_impl_object_id_move` should set the `FL_SEEN_OBJ_ID` flag, so we don't need to do it below.

Reply:
The problem with that is that we're resetting the flags bellow.

Reply:
nvm, I'm an idiot.